Missing POST and DELETE methods for user avatar endpoint

## Problem

The `UserAvatarResource` at `/users/{user_id}/avatar` only implements `GET` method. Upload (POST) and delete (DELETE) methods are missing, causing **405 Method Not Allowed** errors.

## Current State

```python
# app/resources/user_avatar.py
class UserAvatarResource(Resource):
    @require_jwt_auth()
    @check_access_required("read")
    def get(self, user_id):
        # ... GET implementation exists
        
    # ❌ Missing: def post(self, user_id)
    # ❌ Missing: def delete(self, user_id)
```

## Evidence

Integration tests fail with:
```
tests/integration/test_user_avatar_integration.py::test_avatar_upload_to_real_storage
AssertionError: Upload failed: {'message': 'The method is not allowed for the requested URL.'}
assert 405 == 201
```

Storage Service logs show **NO upload requests arrive** - they're rejected by Identity Service routing before reaching Storage.

## Expected Behavior

Follow the same pattern as `CompanyLogoResource` which correctly implements all three methods:

### POST /users/{user_id}/avatar
- Accept multipart/form-data with 'avatar' file
- Validate user exists and JWT matches
- Call `upload_avatar_via_proxy()` from storage_helper
- Update user model: `user.set_avatar(file_id)`
- Return 201 with file_id

### DELETE /users/{user_id}/avatar  
- Validate user exists and JWT matches
- Call `delete_avatar_from_storage()` from storage_helper
- Update user model: `user.remove_avatar()`
- Return 204 No Content

## Reference Implementation

See `app/resources/company_logo.py` lines 35-141 (POST) and 190-242 (DELETE) for working examples.

## Impact

- **All user avatar upload tests fail** (5 integration tests)
- Users cannot upload or delete avatars via API
- Feature is completely non-functional despite User model having `avatar_file_id` and `has_avatar` fields

## Acceptance Criteria

- [ ] Add `post()` method to UserAvatarResource
- [ ] Add `delete()` method to UserAvatarResource  
- [ ] Update OpenAPI spec with POST/DELETE endpoints
- [ ] All 5 avatar integration tests pass
- [ ] Match CompanyLogoResource implementation pattern

## Related

- User model already has avatar support (PR #33)
- Company logo implementation (PR #34) - use as reference
- Integration test infrastructure (current branch)

--- a/app/config.py
+++ b/app/config.py
@@ -50,6 +50,9 @@ class Config:
     STORAGE_SERVICE_URL = os.environ.get("STORAGE_SERVICE_URL")
     STORAGE_REQUEST_TIMEOUT = int(os.environ.get("STORAGE_REQUEST_TIMEOUT", "30"))
 
+    # Maximum avatar file size in MB
+    MAX_AVATAR_SIZE_MB = int(os.environ.get("MAX_AVATAR_SIZE_MB", "5"))
+
     @classmethod
     def validate_storage_config(cls):
         """

--- a/app/resources/organization_unit.py
+++ b/app/resources/organization_unit.py
@@ -21,6 +21,9 @@ from app.models.organization_unit import OrganizationUnit
 from app.schemas.organization_unit_schema import OrganizationUnitSchema
 from app.utils import check_access_required, require_jwt_auth
 
+# Error message constants
+ERROR_INTEGRITY_DUPLICATE = "Integrity error, possibly duplicate entry."
+
 
 class OrganizationUnitListResource(Resource):
     """
@@ -87,7 +90,7 @@ class OrganizationUnitListResource(Resource):
             logger.error(
                 "Integrity error, possibly duplicate entry: %s", str(err)
             )
-            return {"error": "Integrity error, possibly duplicate entry."}, 400
+            return {"error": ERROR_INTEGRITY_DUPLICATE}, 400
         except SQLAlchemyError as err:
             db.session.rollback()
             logger.error("Database error: %s", str(err))
@@ -183,7 +186,7 @@ class OrganizationUnitResource(Resource):
             logger.error(
                 "Integrity error, possibly duplicate entry: %s", str(err)
             )
-            return {"error": "Integrity error, possibly duplicate entry."}, 400
+            return {"error": ERROR_INTEGRITY_DUPLICATE}, 400
         except SQLAlchemyError as err:
             db.session.rollback()
             logger.error("Database error: %s", str(err))
@@ -236,7 +239,7 @@ class OrganizationUnitResource(Resource):
             logger.error(
                 "Integrity error, possibly duplicate entry: %s", str(err)
             )
-            return {"error": "Integrity error, possibly duplicate entry."}, 400
+            return {"error": ERROR_INTEGRITY_DUPLICATE}, 400
         except SQLAlchemyError as err:
             db.session.rollback()
             logger.error("Database error: %s", str(err))

--- a/app/resources/subcontractor.py
+++ b/app/resources/subcontractor.py
@@ -20,6 +20,9 @@ from app.models.subcontractor import Subcontractor
 from app.schemas.subcontractor_schema import SubcontractorSchema
 from app.utils import check_access_required, require_jwt_auth
 
+# Error message constants
+ERROR_SUBCONTRACTOR_NOT_FOUND = "Subcontractor with ID %s not found"
+
 
 class SubcontractorListResource(Resource):
     """
@@ -128,9 +131,9 @@ class SubcontractorResource(Resource):
         subcontractor = Subcontractor.get_by_id(subcontractor_id)
         if not subcontractor:
             logger.warning(
-                "Subcontractor with ID %s not found", subcontractor_id
+                ERROR_SUBCONTRACTOR_NOT_FOUND, subcontractor_id
             )
-            return {"message": "Subcontractor not found"}, 404
+            return {"error": "Subcontractor not found"}, 404
 
         schema = SubcontractorSchema(session=db.session)
         return schema.dump(subcontractor), 200
@@ -162,9 +165,9 @@ class SubcontractorResource(Resource):
             subcontractor = Subcontractor.get_by_id(subcontractor_id)
             if not subcontractor:
                 logger.warning(
-                    "Subcontractor with ID %s not found", subcontractor_id
+                    ERROR_SUBCONTRACTOR_NOT_FOUND, subcontractor_id
                 )
-                return {"message": "Subcontractor not found"}, 404
+                return {"error": "Subcontractor not found"}, 404
 
             updated_subcontractor = subcontractor_schema.load(
                 json_data, instance=subcontractor

--- a/app/resources/subcontractor.py
+++ b/app/resources/subcontractor.py
@@ -22,6 +22,8 @@ from app.utils import check_access_required, require_jwt_auth
 
 # Error message constants
 ERROR_SUBCONTRACTOR_NOT_FOUND = "Subcontractor with ID %s not found"
+MSG_SUBCONTRACTOR_NOT_FOUND = "Subcontractor not found"
+MSG_SUBCONTRACTOR_DELETED = "Subcontractor deleted successfully"
 
 
 class SubcontractorListResource(Resource):
@@ -133,7 +135,7 @@ class SubcontractorResource(Resource):
             logger.warning(
                 ERROR_SUBCONTRACTOR_NOT_FOUND, subcontractor_id
             )
-            return {"error": "Subcontractor not found"}, 404
+            return {"error": MSG_SUBCONTRACTOR_NOT_FOUND}, 404
 
         schema = SubcontractorSchema(session=db.session)
         return schema.dump(subcontractor), 200
@@ -167,7 +169,7 @@ class SubcontractorResource(Resource):
                 logger.warning(
                     ERROR_SUBCONTRACTOR_NOT_FOUND, subcontractor_id
                 )
-                return {"error": "Subcontractor not found"}, 404
+                return {"error": MSG_SUBCONTRACTOR_NOT_FOUND}, 404
 
             updated_subcontractor = subcontractor_schema.load(
                 json_data, instance=subcontractor
@@ -217,9 +219,9 @@ class SubcontractorResource(Resource):
             subcontractor = Subcontractor.get_by_id(subcontractor_id)
             if not subcontractor:
                 logger.warning(
-                    "Subcontractor with ID %s not found", subcontractor_id
+                    ERROR_SUBCONTRACTOR_NOT_FOUND, subcontractor_id
                 )
-                return {"message": "Subcontractor not found"}, 404
+                return {"message": MSG_SUBCONTRACTOR_NOT_FOUND}, 404
 
             updated_subcontractor = subcontractor_schema.load(
                 json_data, instance=subcontractor, partial=True
@@ -254,14 +256,14 @@ class SubcontractorResource(Resource):
         subcontractor = Subcontractor.get_by_id(subcontractor_id)
         if not subcontractor:
             logger.warning(
-                "Subcontractor with ID %s not found", subcontractor_id
+                ERROR_SUBCONTRACTOR_NOT_FOUND, subcontractor_id
             )
-            return {"message": "Subcontractor not found"}, 404
+            return {"message": MSG_SUBCONTRACTOR_NOT_FOUND}, 404
 
         try:
             db.session.delete(subcontractor)
             db.session.commit()
-            return {"message": "Subcontractor deleted successfully"}, 204
+            return {"message": MSG_SUBCONTRACTOR_DELETED}, 204
         except SQLAlchemyError as e:
             db.session.rollback()
             logger.error("Database error: %s", str(e))

--- a/app/resources/user.py
+++ b/app/resources/user.py
@@ -36,6 +36,9 @@ from app.storage_helper import (
 )
 from app.utils import check_access_required, require_jwt_auth
 
+# Content type constants
+CONTENT_TYPE_MULTIPART = "multipart/form-data"
+
 
 # Helper functions for user operations
 def _validate_avatar_url_field(avatar_url_value, context="POST"):
@@ -51,7 +54,7 @@ def _validate_avatar_url_field(avatar_url_value, context="POST"):
         f"Is base64 data URI: {is_base64}, "
         f"Preview: {avatar_url_value[:50]}... "
         f"EXPECTED: Frontend should send file via "
-        f"'avatar' field in multipart/form-data, "
+        f"'avatar' field in {CONTENT_TYPE_MULTIPART}, "
         f"NOT avatar_url. The backend manages avatar_url "
         f"after upload to Storage Service."
     )
@@ -196,26 +199,44 @@ def _parse_patch_request_data():
         f"[PATCH] Has form: {bool(request.form)}, Has files: {bool(request.files)}"
     )
 
-    # Handle multipart/form-data or JSON
+    # Check if request has multipart data
     has_form_data = len(request.form) > 0 or len(request.files) > 0
     is_multipart_content = (
         request.content_type and "multipart" in request.content_type
     ) or (request.mimetype and "multipart" in request.mimetype)
 
     if has_form_data or is_multipart_content:
-        json_data = _parse_multipart_data_with_debug(context="PATCH")
-    else:
-        # Try to parse as JSON
-        json_data = request.get_json(force=True, silent=True) or {}
-        logger.debug(f"[PATCH] Parsed JSON data: {bool(json_data)}")
+        return _parse_multipart_data_with_debug(context="PATCH")
 
-        # Remove avatar_url from JSON data if present
-        if "avatar_url" in json_data:
-            _validate_avatar_url_field(json_data["avatar_url"], "PATCH")
-            json_data.pop("avatar_url", None)
+    # Parse as JSON
+    json_data = request.get_json(force=True, silent=True) or {}
+    logger.debug(f"[PATCH] Parsed JSON data: {bool(json_data)}")
+
+    # Remove avatar_url from JSON data if present
+    if "avatar_url" in json_data:
+        _validate_avatar_url_field(json_data["avatar_url"], "PATCH")
+        json_data.pop("avatar_url", None)
 
     logger.debug(f"[PATCH] json_data after filtering: {json_data}")
     return json_data
+
+
+def _process_avatar_upload_for_patch(user, company_id):
+    """
+    Handle avatar upload for PATCH operation.
+
+    Returns:
+        str or None: File ID if avatar was uploaded, None otherwise.
+
+    Raises:
+        ValidationError: If upload fails.
+    """
+    try:
+        return _handle_avatar_upload_for_update(user, company_id)
+    except AvatarValidationError as e:
+        raise ValidationError({"avatar": [str(e)]}) from e
+    except StorageServiceError as exc:
+        raise ValidationError({"avatar": ["Failed to upload avatar"]}) from exc
 
 
 class UserListResource(Resource):
@@ -279,8 +300,8 @@ class UserListResource(Resource):
         # Handle multipart/form-data or JSON
         is_multipart = (
             request.content_type
-            and "multipart/form-data" in request.content_type
-        ) or (request.mimetype and "multipart/form-data" in request.mimetype)
+            and CONTENT_TYPE_MULTIPART in request.content_type
+        ) or (request.mimetype and CONTENT_TYPE_MULTIPART in request.mimetype)
 
         # Parse request data
         json_data = _parse_request_data(is_multipart, context="POST")
@@ -395,8 +416,8 @@ class UserResource(Resource):
         # Handle multipart/form-data or JSON
         is_multipart = (
             request.content_type
-            and "multipart/form-data" in request.content_type
-        ) or (request.mimetype and "multipart/form-data" in request.mimetype)
+            and CONTENT_TYPE_MULTIPART in request.content_type
+        ) or (request.mimetype and CONTENT_TYPE_MULTIPART in request.mimetype)
 
         # Parse request data
         json_data = _parse_request_data(is_multipart, context="PUT")
@@ -495,15 +516,9 @@ class UserResource(Resource):
 
         try:
             # Handle avatar upload if present
-            uploaded_file_id = None
-            try:
-                uploaded_file_id = _handle_avatar_upload_for_update(
-                    user, company_id
-                )
-            except AvatarValidationError as e:
-                return {"message": str(e)}, 400
-            except StorageServiceError:
-                return {"message": "Failed to upload avatar"}, 500
+            uploaded_file_id = _process_avatar_upload_for_patch(
+                user, company_id
+            )
 
             # Company_id modification is prevented by schema validation
             logger.debug(f"[PATCH] json_data before schema.load: {json_data}")

--- a/app/storage_helper.py
+++ b/app/storage_helper.py
@@ -6,30 +6,16 @@ Helper functions for interacting with the Storage Service.
 Provides avatar upload/delete functionality for the Identity Service.
 """
 
-import os
-
 import requests
 
 from app.logger import logger
 
-# Configuration
-STORAGE_SERVICE_URL = os.getenv(
-    "STORAGE_SERVICE_URL", "http://storage-service:5000"
-)
-REQUEST_TIMEOUT = int(os.getenv("STORAGE_REQUEST_TIMEOUT", "30"))
-MAX_AVATAR_SIZE = int(os.getenv("MAX_AVATAR_SIZE_MB", "5")) * 1024 * 1024
-
-# Storage Service integration toggle
-USE_STORAGE_SERVICE = os.getenv("USE_STORAGE_SERVICE", "true").lower() in (
-    "true",
-    "yes",
-    "1",
-)
-
 
 def is_storage_service_enabled() -> bool:
     """Check if Storage Service integration is enabled."""
-    return USE_STORAGE_SERVICE
+    from flask import current_app  # pylint: disable=import-outside-toplevel
+
+    return current_app.config.get("USE_STORAGE_SERVICE", False)
 
 
 # Allowed MIME types for avatars
@@ -51,7 +37,7 @@ class AvatarValidationError(Exception):
 
 
 def validate_avatar(
-    file_data: bytes, content_type: str, max_size: int = MAX_AVATAR_SIZE
+    file_data: bytes, content_type: str, max_size: int = None
 ) -> None:
     """
     Validate an avatar file.
@@ -59,11 +45,16 @@ def validate_avatar(
     Args:
         file_data: Binary file data
         content_type: MIME type of the file
-        max_size: Maximum size in bytes
+        max_size: Maximum size in bytes (default from config)
 
     Raises:
         AvatarValidationError: If validation fails
     """
+    from flask import current_app  # pylint: disable=import-outside-toplevel
+
+    if max_size is None:
+        max_size = current_app.config.get("MAX_AVATAR_SIZE_MB", 5) * 1024 * 1024
+
     if not file_data:
         logger.error("Avatar file is empty")
         raise AvatarValidationError("Avatar file is empty")
@@ -178,7 +169,9 @@ def upload_avatar_via_proxy(  # pylint: disable=too-many-locals
         user_id, company_id, file_data, content_type, filename
     )
 
-    url = f"{STORAGE_SERVICE_URL}/upload/proxy"
+    from flask import current_app  # pylint: disable=import-outside-toplevel
+
+    url = f"{current_app.config['STORAGE_SERVICE_URL']}/upload/proxy"
     logger.debug(f"Uploading avatar for user {user_id} to {url}")
 
     try:
@@ -186,12 +179,13 @@ def upload_avatar_via_proxy(  # pylint: disable=too-many-locals
             f"Uploading avatar for user {user_id}: {len(file_data)} bytes"
         )
 
+        timeout = current_app.config.get("STORAGE_REQUEST_TIMEOUT", 30)
         response = requests.post(
             url,
             files=files,
             data=data,
             headers=headers,
-            timeout=REQUEST_TIMEOUT,
+            timeout=timeout,
         )
 
         logger.debug(
@@ -266,7 +260,9 @@ def delete_avatar(user_id: str, company_id: str, file_id: str) -> None:
         logger.warning("No file_id provided for deletion")
         return
 
-    url = f"{STORAGE_SERVICE_URL}/delete"
+    from flask import current_app  # pylint: disable=import-outside-toplevel
+
+    url = f"{current_app.config['STORAGE_SERVICE_URL']}/delete"
 
     headers = {
         "X-User-ID": user_id,
@@ -283,8 +279,9 @@ def delete_avatar(user_id: str, company_id: str, file_id: str) -> None:
     try:
         logger.info(f"Deleting avatar with file_id: {file_id}")
 
+        timeout = current_app.config.get("STORAGE_REQUEST_TIMEOUT", 30)
         response = requests.delete(
-            url, json=payload, headers=headers, timeout=REQUEST_TIMEOUT
+            url, json=payload, headers=headers, timeout=timeout
         )
 
         # Accept both 200 and 404 (already deleted)
@@ -330,7 +327,9 @@ def create_user_directories(user_id: str, company_id: str) -> None:
         )
         return
 
-    url = f"{STORAGE_SERVICE_URL}/upload/proxy"
+    from flask import current_app  # pylint: disable=import-outside-toplevel
+
+    url = f"{current_app.config['STORAGE_SERVICE_URL']}/upload/proxy"
 
     headers = {
         "X-User-ID": user_id,
@@ -361,12 +360,13 @@ def create_user_directories(user_id: str, company_id: str) -> None:
                 f"Creating directory marker: {logical_path} for user {user_id}"
             )
 
+            timeout = current_app.config.get("STORAGE_REQUEST_TIMEOUT", 30)
             response = requests.post(
                 url,
                 files=files,
                 data=data,
                 headers=headers,
-                timeout=REQUEST_TIMEOUT,
+                timeout=timeout,
             )
 
             response.raise_for_status()
@@ -415,8 +415,10 @@ def delete_user_storage(user_id: str, company_id: str) -> None:
     # Since the Storage Service doesn't have a "delete directory" endpoint,
     # we need to list all files and delete them one by one
 
+    from flask import current_app  # pylint: disable=import-outside-toplevel
+
     # First, try to list all files for this user
-    list_url = f"{STORAGE_SERVICE_URL}/list"
+    list_url = f"{current_app.config['STORAGE_SERVICE_URL']}/list"
 
     headers = {
         "X-User-ID": user_id,
@@ -432,8 +434,9 @@ def delete_user_storage(user_id: str, company_id: str) -> None:
     try:
         logger.info(f"Listing files for user {user_id} to delete")
 
+        timeout = current_app.config.get("STORAGE_REQUEST_TIMEOUT", 30)
         response = requests.get(
-            list_url, params=params, headers=headers, timeout=REQUEST_TIMEOUT
+            list_url, params=params, headers=headers, timeout=timeout
         )
 
         if response.status_code == 404:
@@ -447,7 +450,7 @@ def delete_user_storage(user_id: str, company_id: str) -> None:
         logger.info(f"Found {len(files)} files to delete for user {user_id}")
 
         # Delete each file
-        delete_url = f"{STORAGE_SERVICE_URL}/delete"
+        delete_url = f"{current_app.config['STORAGE_SERVICE_URL']}/delete"
 
         for file_meta in files:
             file_id = file_meta.get("file_id")
@@ -464,7 +467,7 @@ def delete_user_storage(user_id: str, company_id: str) -> None:
                     delete_url,
                     json=delete_payload,
                     headers=headers,
-                    timeout=REQUEST_TIMEOUT,
+                    timeout=timeout,
                 )
 
                 if del_response.status_code in (200, 204, 404):

--- a/docker-compose.integration.yml
+++ b/docker-compose.integration.yml
@@ -1,5 +1,3 @@
-version: '3.8'
-
 services:
   minio:
     image: minio/minio:latest
@@ -29,11 +27,18 @@ services:
       LOG_LEVEL: DEBUG
       DATABASE_URL: sqlite:///dev.db
       JWT_SECRET: integration-test-secret-key
-      IDENTITY_SERVICE_URL: http://host.docker.internal:5002/
+      # Disable project service integration for standalone testing
+      USE_PROJECT_SERVICE: "false"
+      # MinIO configuration
       MINIO_SERVICE_URL: http://minio:9000/
       MINIO_ACCESS_KEY: minioadmin
       MINIO_SECRET_KEY: minioadmin123
       MINIO_BUCKET_PREFIX: integration-test
+      # Docker-specific flags
+      IN_DOCKER_CONTAINER: "1"
+      WAIT_FOR_DB: "false"
+      RUN_MIGRATIONS: "true"
+      APP_MODE: "development"
     depends_on:
       minio:
         condition: service_healthy

--- a/run-integration-tests.sh
+++ b/run-integration-tests.sh
@@ -20,21 +20,21 @@ echo "======================================"
 cleanup() {
     echo ""
     echo "🧹 Cleaning up services..."
-    docker-compose -f "$COMPOSE_FILE" down -v
+    docker compose -f "$COMPOSE_FILE" down -v
 }
 
 # Trap EXIT to ensure cleanup
 trap cleanup EXIT
 
 # Check if services are already running
-if docker-compose -f "$COMPOSE_FILE" ps | grep -q "Up"; then
+if docker compose -f "$COMPOSE_FILE" ps | grep -q "Up"; then
     echo "⚠️  Services already running. Stopping..."
-    docker-compose -f "$COMPOSE_FILE" down
+    docker compose -f "$COMPOSE_FILE" down
 fi
 
 # Start services
 echo "🚀 Starting MinIO and Storage Service..."
-docker-compose -f "$COMPOSE_FILE" up -d
+docker compose -f "$COMPOSE_FILE" up -d
 
 # Wait for services to be healthy
 echo "⏳ Waiting for services to be healthy..."
@@ -42,9 +42,9 @@ timeout=60
 elapsed=0
 
 while [ $elapsed -lt $timeout ]; do
-    if docker-compose -f "$COMPOSE_FILE" ps | grep -q "healthy"; then
-        minio_health=$(docker-compose -f "$COMPOSE_FILE" ps minio | grep -c "healthy" || echo "0")
-        storage_health=$(docker-compose -f "$COMPOSE_FILE" ps storage-service | grep -c "healthy" || echo "0")
+    if docker compose -f "$COMPOSE_FILE" ps | grep -q "healthy"; then
+        minio_health=$(docker compose -f "$COMPOSE_FILE" ps minio | grep -c "healthy" || echo "0")
+        storage_health=$(docker compose -f "$COMPOSE_FILE" ps storage-service | grep -c "healthy" || echo "0")
         
         if [ "$minio_health" -ge 1 ] && [ "$storage_health" -ge 1 ]; then
             echo "✅ All services are healthy!"
@@ -61,17 +61,24 @@ if [ $elapsed -ge $timeout ]; then
     echo "❌ Services failed to become healthy within $timeout seconds"
     echo ""
     echo "Service status:"
-    docker-compose -f "$COMPOSE_FILE" ps
+    docker compose -f "$COMPOSE_FILE" ps
     echo ""
     echo "Logs:"
-    docker-compose -f "$COMPOSE_FILE" logs
+    docker compose -f "$COMPOSE_FILE" logs
     exit 1
 fi
 
 # Show service status
 echo ""
 echo "📊 Service Status:"
-docker-compose -f "$COMPOSE_FILE" ps
+docker compose -f "$COMPOSE_FILE" ps
+
+# Export environment variables for integration tests
+export STORAGE_SERVICE_URL="http://localhost:5001"
+export USE_STORAGE_SERVICE="true"
+export JWT_SECRET="integration-test-secret-key"
+export DATABASE_URL="sqlite:///:memory:"
+export FLASK_ENV="testing"
 
 # Run integration tests
 echo ""
@@ -93,7 +100,7 @@ if [ $exit_code -ne 0 ]; then
     echo ""
     echo "📜 Service Logs:"
     echo "======================================"
-    docker-compose -f "$COMPOSE_FILE" logs --tail=50
+    docker compose -f "$COMPOSE_FILE" logs --tail=50
 fi
 
 exit $exit_code

--- a/sonar-project.properties
+++ b/sonar-project.properties
@@ -1,0 +1,26 @@
+# SonarCloud Configuration for Identity Service
+sonar.projectKey=bengeek06_identity-api-waterfall
+sonar.organization=bengeek06
+
+# Project metadata
+sonar.projectName=Identity API Waterfall
+sonar.projectVersion=1.0
+
+# Python version
+sonar.python.version=3.13
+
+# Source directories
+sonar.sources=app
+sonar.tests=tests
+
+# Exclusions
+sonar.exclusions=**/migrations/**,**/__pycache__/**,**/venv/**,**/instance/**
+
+# Test exclusions
+sonar.test.exclusions=tests/**
+
+# Coverage (if you have coverage reports)
+# sonar.python.coverage.reportPaths=coverage.xml
+
+# Encoding
+sonar.sourceEncoding=UTF-8

--- a/tests/integration/conftest.py
+++ b/tests/integration/conftest.py
@@ -10,6 +10,7 @@ import uuid
 
 import pytest
 import requests
+from werkzeug.security import generate_password_hash
 
 from app import create_app
 from app.models import db
@@ -48,7 +49,7 @@ def check_services_health(integration_config):
             response.status_code == 200
         ), f"Storage Service unhealthy: {response.status_code}"
         print(f"✓ Storage Service healthy at {storage_url}")
-    except Exception as e:
+    except (requests.exceptions.RequestException, AssertionError) as e:
         pytest.skip(
             f"Storage Service not available at {storage_url}: {e}\n"
             "Run: docker-compose -f docker-compose.integration.yml up -d"
@@ -56,9 +57,13 @@ def check_services_health(integration_config):
 
 
 @pytest.fixture
-def integration_app(integration_config, check_services_health):
+def integration_app(integration_config, check_services_health):  # pylint: disable=unused-argument
     """
     Flask app configured for integration testing with real Storage Service.
+
+    Args:
+        integration_config: Configuration dictionary for integration tests
+        check_services_health: Fixture that verifies services are healthy (implicit dependency)
     """
     # Set environment variables for real services
     os.environ["USE_STORAGE_SERVICE"] = "true"
@@ -112,8 +117,6 @@ def real_user(integration_session, real_company):
     """
     Create a real user for integration tests.
     """
-    from werkzeug.security import generate_password_hash
-    
     user = User(
         id=str(uuid.uuid4()),
         email="integration@test.com",
@@ -144,6 +147,8 @@ def storage_api_client(integration_config):
     """
 
     class StorageAPIClient:
+        """Helper class for making authenticated requests to Storage Service API."""
+
         def __init__(self, base_url):
             self.base_url = base_url
 

--- a/tests/integration/conftest.py
+++ b/tests/integration/conftest.py
@@ -112,14 +112,16 @@ def real_user(integration_session, real_company):
     """
     Create a real user for integration tests.
     """
+    from werkzeug.security import generate_password_hash
+    
     user = User(
         id=str(uuid.uuid4()),
         email="integration@test.com",
         first_name="Integration",
         last_name="Test",
         company_id=real_company.id,
+        hashed_password=generate_password_hash("integration-password"),
     )
-    user.set_password("integration-password")
     integration_session.add(user)
     integration_session.commit()
     yield user

--- a/tests/integration/test_user_avatar_integration.py
+++ b/tests/integration/test_user_avatar_integration.py
@@ -226,11 +226,11 @@ def test_avatar_isolation_between_users(
     Test that avatars are properly isolated between users.
     User A should not be able to access User B's avatar.
     """
-    from app.models.user import User
+    from werkzeug.security import generate_password_hash  # pylint: disable=import-outside-toplevel
+
+    from app.models.user import User  # pylint: disable=import-outside-toplevel
 
     # Create second user
-    from werkzeug.security import generate_password_hash
-    
     user_b = User(
         email="userb@integration.test",
         first_name="User",

--- a/tests/integration/test_user_avatar_integration.py
+++ b/tests/integration/test_user_avatar_integration.py
@@ -229,13 +229,15 @@ def test_avatar_isolation_between_users(
     from app.models.user import User
 
     # Create second user
+    from werkzeug.security import generate_password_hash
+    
     user_b = User(
         email="userb@integration.test",
         first_name="User",
         last_name="B",
         company_id=real_company.id,
+        hashed_password=generate_password_hash("password"),
     )
-    user_b.set_password("password")
     integration_session.add(user_b)
     integration_session.commit()
 

